qiskit_experiments/framework/composite/composite_analysis.py

```diff
@@ -104,6 +105,12 @@ def _wait_for_components(experiment_data, component_ids):
 
         experiment_data.add_analysis_callback(_wait_for_components, component_ids=component_ids)
 
+        # TODO implement proper locking mechanism.
+        # Because blocking mechanism is sometime terminated before above callback is added to
+        # the experiment_data callback queue list, we need to extend execution time of this routine
+        # to ensure this will complete after the callback is actually queued.
+        sleep(1)
```

Comment:
I dont think this problem is specific to composite experiment. Returning before the callback has updated the status could happen to any experiment and probably needs some tweaks in DbExperimentData.status. Adding a 1 second sleep here seems not a good solution.